samples/java-protobuf-doc-snippets/src/main/java/com/example/DelegatingServiceAction.java

```diff
@@ -37,4 +38,29 @@ public Effect<DelegatingServiceApi.Result> addAndReturn(DelegatingServiceApi.Req
     return effects().asyncReply(result);  // <5>
   }
   // end::delegating-action[]
+
+  public Effect<DelegatingServiceApi.Result> addAndReturnLifted(DelegatingServiceApi.Request request) {
+    CounterService counterService = actionContext().getGrpcClient(CounterService.class, "counter");
+    // tag::delegating-action-lifted[]
+    SingleResponseRequestBuilder<CounterApi.IncreaseValue, Empty> counterServiceClientBuilder
+            = ((CounterServiceClient) counterService).increase().addHeader("key", "value"); // <1>
+
+
+    CounterApi.IncreaseValue increaseValue = CounterApi.IncreaseValue.newBuilder()
+            .setCounterId(request.getCounterId())
+            .setValue(1)
+            .build();
+    CompletionStage<Empty> increaseCompleted = counterServiceClientBuilder.invoke(increaseValue);  // <2>
```

Comment:
I'd say the canonical use is to never catch the builder, avoiding the type soup of the builder

```
CompletionStage<Empty> increaseCompleted = 
    ((CounterServiceClient) counterService).increase()
        .addHeader("key", "value") // <1>
        .invoke(increaseValue); // <2>

Reply:
I wasn't sure. I wanted the user to know about that class. I think it's clearer in that respect, what's happening underneath. But it's true that it's a bit messy. 
I can use the canonical/cleaner way, np.

Reply:
done